pkg/codegen/schema/loader.go

```diff
@@ -333,6 +333,12 @@ func (l *pluginLoader) loadSchemaBytes(
 	if err != nil {
 		return nil, nil, err
 	}
+	// note that in the case of parameterization, parameterizing can change
+	// the schema and expected version, eg in the terraform provider.
+	//
+	// The loader also compares the version against the schema against this, so
+	// they must be updated to be referencing the same semantic package (ie the
+	// parameterized package, see [LoadPackageReferenceV2])
```

Comment:
This comment is much clearer; thanks for working on it. I think the icing on the cake would be an example in it, e.g. "we are given a descriptor ...{ name = "terraform-provider", version = "...", parameterization = ... } which results in a schema ...; if we don't adjust the version based on the parameterization yada yada yada"